pkg/apis/traceflow/v1/types.go

```diff
@@ -0,0 +1,162 @@
+	ERROR   Phase = "ERROR"
+)
+
+type Component string
+
+const (
+	SPOOFGUARD    Component = "SPOOFGUARD"
+	LB            Component = "LB"
+	ROUTING       Component = "ROUTING"
+	NETWORKPOLICY Component = "NETWORKPOLICY"
+	FORWARDING    Component = "FORWARDING"
+)
+
+type Action string
+
+const (
+	DELIVERED Action = "DELIVERED"
+	RECEIVED  Action = "RECEIVED"
+	FORWARDED Action = "FORWARDED"
+	DROPPED   Action = "DROPPED"
+)
+
+type PacketResouceType string
+
+const (
+	FieldsPacketData PacketResouceType = "FieldsPacketData"
+)
+
+type PacketTransportType string
+
+const (
+	UNICAST PacketTransportType = "UNICAST"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type Traceflow struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Source      `json:"source"`
+	Destination `json:"destination"`
+	RoundID     string `json:"roundID,omitempty"`
+
+	Packet `json:"packet"`
+	Status `json:"status"`
+}
+
+type Source struct {
+	SrcNamespace string `json:"srcNamespace,omitempty"`
+	SrcPod       string `json:"srcPod,omitempty"`
+}
+
+type Destination struct {
+	DstNamespace string `json:"dstNamespace,omitempty"`
+	DstPod       string `json:"dstPod,omitempty"`
+	DstService   string `json:"dstService,omitempty"`
+	DstIP        string `json:"dstIP,omitempty"`
+}
+
+type IPHeader struct {
+	SrcIP    string `json:"srcIP,omitempty"`
+	Protocol int32  `json:"protocol,omitempty"`
+	TTL      int32  `json:"ttl,omitempty"`
+	Flags    int32  `json:"flags,omitempty"`
+}
+
+type TransportHeader struct {
+	ICMPEchoRequestHeader `json:"icmpEchoRequestHeader"`
+	UDPHeader             `json:"udpHeader"`
+	TCPHeader             `json:"tcpHeader"`
+}
+
+type ICMPEchoRequestHeader struct {
+	ID       int32 `json:"id,omitempty"`
+	Sequence int32 `json:"sequence,omitempty"`
+}
+
+type UDPHeader struct {
+	SrcUDPPort int32 `json:"srcUDPPort,omitempty"`
+	DstUDPPort int32 `json:"dstUDPPort,omitempty"`
+}
+
+type TCPHeader struct {
+	SrcTCPPort int32 `json:"srcTCPPort,omitempty"`
+	DstTCPPort int32 `json:"dstTCPPort,omitempty"`
+	TCPFlags   int32 `json:"tcpFlags,omitempty"`
+}
+
+type Packet struct {
+	ResouceType   PacketResouceType   `json:"resouceType,omitempty"`
+	TransportType PacketTransportType `json:"transportType,omitempty"`
+	PayloadString string              `json:"payloadString,omitempty"`
+	PayloadSize   int32               `json:"payloadSize,omitempty"`
+
+	IPHeader        `json:"ipHeader"`
+	TransportHeader `json:"transportHeader"`
+}
+
+type Status struct {
+	Phase        Phase         `json:"phase,omitempty"`
+	CrossNodeTag uint8         `json:"crossNodeTag,omitempty"`
+	NodeSender   []Observation `json:"nodeSender,omitempty"`
+	NodeReceiver []Observation `json:"nodeReceiver,omitempty"`
+}
+
+type Observation struct {
+	Component       Component  `json:"component,omitempty"`
+	SubComponent    string     `json:"subComponent,omitempty"`
```

Comment:
Do we have subcomponents defined already? If no subcomponent is supported, probably remove it for now. Could always add more fields later.

Reply:
Yes, removed it.